### ISSUE TITLE
Investigate available camera capabilities we can leverage

### ISSUE THREAD

**Issue by azasypkin** [project-lighthouse/camera-app.ios#47]
@sfoster @punamdahiya feel free to add ideas and capabilities you think would be useful to have.

I see if we can rely on something that tells us that we have `an object` in focus, not blurred etc. Otherwise maybe I'll check if we can use OpenCV for that (eg. detect contours, extract keypoints and see if we have enough of them).

**Comment by azasypkin** [comment]
https://developer.apple.com/library/content/documentation/DeviceInformation/Reference/iOSDeviceCompatibility/Cameras/Cameras.html

Strangely (or obviously since we're talking about Apple) iPhone 5(s) cameras aren't mentioned.

**Comment by azasypkin** [comment]
Camera properties we can control directly from C++ (VideoCapture): https://github.com/opencv/opencv/blob/master/modules/videoio/include/opencv2/videoio.hpp#L486-L491

**Comment by azasypkin** [comment]
Okay here is [the list of all currently available frame metadata that iOS can give us](https://developer.xamarin.com/api/type/AVFoundation.AVMetadataObjectType/). Basically `face`, `none` and a bunch of machine readable codes (that we can use later on):

* AztecCode
* Code128Code
* Code39Code
* Code39Mod43Code
* Code93Code
* DataMatrixCode
* EAN13Code
* EAN8Code
* __Face__
* Interleaved2of5Code
* ITF14Code
* __None__
* PDF417Code
* QRCode
* UPCECode

**Comment by sfoster** [comment]
That list means nothing to me. Can you summarize the significance of these metadata properties @azasypkin

**Comment by azasypkin** [comment]
>That list means nothing to me. 

Well the summary is pretty short - there is nothing useful for us at this stage in iOS framework, we'll have to rely on *something* based on OpenCV. 

AVFoundationFramework can only give us *additional metadata* about _faces_ or _machine readable codes_ it sees in the photo (location, bounding rect, roll/yaw angle and type for the codes). We may need it later though, but it's too early to think about it.
